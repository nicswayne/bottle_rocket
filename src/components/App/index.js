import React from 'react';
import { Route, Switch } from 'react-router-dom';

import List from '../List';
import Web from '../Web';

import '../../styles/App.css';

const App = () => {
    return (
        <div>
            <Switch>
                <Route path="/web" component={Web} />
                <Route path="/" component={List} />
            </Switch>
        </div>
    );
};

export default App;
