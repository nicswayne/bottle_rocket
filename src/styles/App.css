@charset "UTF-8";
/**
 * Foundation for Sites by ZURB
 * Version 6.3.1
 * foundation.zurb.com
 * Licensed under MIT Open Source
 */
@import url("https://fonts.googleapis.com/css?family=Nunito:400,600");
/*! normalize-scss | MIT/GPLv2 License | bit.ly/normalize-scss */
/* Document
       ========================================================================== */
/**
     * 1. Change the default font family in all browsers (opinionated).
     * 2. Correct the line height in all browsers.
     * 3. Prevent adjustments of font size after orientation changes in
     *    IE on Windows Phone and in iOS.
     */
html {
  font-family: sans-serif;
  /* 1 */
  line-height: 1.15;
  /* 2 */
  -ms-text-size-adjust: 100%;
  /* 3 */
  -webkit-text-size-adjust: 100%;
  /* 3 */
}

/* Sections
       ========================================================================== */
/**
     * Remove the margin in all browsers (opinionated).
     */
body {
  margin: 0;
}

/**
     * Add the correct display in IE 9-.
     */
article,
aside,
footer,
header,
nav,
section {
  display: block;
}

/**
     * Correct the font size and margin on `h1` elements within `section` and
     * `article` contexts in Chrome, Firefox, and Safari.
     */
h1 {
  font-size: 2em;
  margin: 0.67em 0;
}

/* Grouping content
       ========================================================================== */
/**
     * Add the correct display in IE 9-.
     */
figcaption,
figure {
  display: block;
}

/**
     * Add the correct margin in IE 8.
     */
figure {
  margin: 1em 40px;
}

/**
     * 1. Add the correct box sizing in Firefox.
     * 2. Show the overflow in Edge and IE.
     */
hr {
  box-sizing: content-box;
  /* 1 */
  height: 0;
  /* 1 */
  overflow: visible;
  /* 2 */
}

/**
     * Add the correct display in IE.
     */
main {
  display: block;
}

/**
     * 1. Correct the inheritance and scaling of font size in all browsers.
     * 2. Correct the odd `em` font sizing in all browsers.
     */
pre {
  font-family: monospace, monospace;
  /* 1 */
  font-size: 1em;
  /* 2 */
}

/* Links
       ========================================================================== */
/**
     * 1. Remove the gray background on active links in IE 10.
     * 2. Remove gaps in links underline in iOS 8+ and Safari 8+.
     */
a {
  background-color: transparent;
  /* 1 */
  -webkit-text-decoration-skip: objects;
  /* 2 */
}

/**
     * Remove the outline on focused links when they are also active or hovered
     * in all browsers (opinionated).
     */
a:active,
a:hover {
  outline-width: 0;
}

/* Text-level semantics
       ========================================================================== */
/**
     * 1. Remove the bottom border in Firefox 39-.
     * 2. Add the correct text decoration in Chrome, Edge, IE, Opera, and Safari.
     */
abbr[title] {
  border-bottom: none;
  /* 1 */
  text-decoration: underline;
  /* 2 */
  text-decoration: underline dotted;
  /* 2 */
}

/**
     * Prevent the duplicate application of `bolder` by the next rule in Safari 6.
     */
b,
strong {
  font-weight: inherit;
}

/**
     * Add the correct font weight in Chrome, Edge, and Safari.
     */
b,
strong {
  font-weight: bolder;
}

/**
     * 1. Correct the inheritance and scaling of font size in all browsers.
     * 2. Correct the odd `em` font sizing in all browsers.
     */
code,
kbd,
samp {
  font-family: monospace, monospace;
  /* 1 */
  font-size: 1em;
  /* 2 */
}

/**
     * Add the correct font style in Android 4.3-.
     */
dfn {
  font-style: italic;
}

/**
     * Add the correct background and color in IE 9-.
     */
mark {
  background-color: #ff0;
  color: #000;
}

/**
     * Add the correct font size in all browsers.
     */
small {
  font-size: 80%;
}

/**
     * Prevent `sub` and `sup` elements from affecting the line height in
     * all browsers.
     */
sub,
sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}

sub {
  bottom: -0.25em;
}

sup {
  top: -0.5em;
}

/* Forms
       ========================================================================== */
/**
     * 1. Change the font styles in all browsers (opinionated).
     * 2. Remove the margin in Firefox and Safari.
     */
button,
input,
optgroup,
select,
textarea {
  font-family: sans-serif;
  /* 1 */
  font-size: 100%;
  /* 1 */
  line-height: 1.15;
  /* 1 */
  margin: 0;
  /* 2 */
}

/**
     * Show the overflow in IE.
     */
button {
  overflow: visible;
}

/**
     * Remove the inheritance of text transform in Edge, Firefox, and IE.
     * 1. Remove the inheritance of text transform in Firefox.
     */
button,
select {
  /* 1 */
  text-transform: none;
}

/**
     * 1. Prevent a WebKit bug where (2) destroys native `audio` and `video`
     *    controls in Android 4.
     * 2. Correct the inability to style clickable types in iOS and Safari.
     */
button,
html [type="button"],
[type="reset"],
[type="submit"] {
  -webkit-appearance: button;
  /* 2 */
}

button,
[type="button"],
[type="reset"],
[type="submit"] {
  /**
       * Remove the inner border and padding in Firefox.
       */
  /**
       * Restore the focus styles unset by the previous rule.
       */
}
button::-moz-focus-inner,
[type="button"]::-moz-focus-inner,
[type="reset"]::-moz-focus-inner,
[type="submit"]::-moz-focus-inner {
  border-style: none;
  padding: 0;
}
button:-moz-focusring,
[type="button"]:-moz-focusring,
[type="reset"]:-moz-focusring,
[type="submit"]:-moz-focusring {
  outline: 1px dotted ButtonText;
}

/**
     * Show the overflow in Edge.
     */
input {
  overflow: visible;
}

img {
  display: inline-block;
  vertical-align: middle;
  max-width: 100%;
  height: auto;
  -ms-interpolation-mode: bicubic;
}

button {
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  border-radius: 0;
  background: transparent;
  line-height: 1;
}

/************ TEXT *************/

.text {
  color: #fff;
}

.header .text,
.item .name,
.details-header .name {
  font-family: "Nunito";
  font-weight: 600;
}

.item .category,
.details-header .category,
.details {
  font-family: "Nunito";
  font-weight: 400;
}

.details .address {
  font-size: 16px;
}

/************* HEADER *************/

.header {
  background-color: #43e895;
  text-align: center;
  height: 70px;
  position: fixed;
  width: 100%;
}

.header .text {
  font-size: 17px;
  line-height: 90px;
}

.header .map {
  height: 35px;
  padding-right: 12px;
  padding-top: 26px;
  float: right;
}

/************* LIST *************/

.list-body {
  margin-bottom: 50px;
  display: flex;
}

.item {
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;
  box-shadow: inset 0px -135px 135px -50px rgba(0, 0, 0, 8);
  -webkit-box-shadow: inset 0px -135px 135px -50px rgba(0, 0, 0, 0.8);
  justify-content: flex-end;
  display: flex;
  flex-direction: column;
  height: 180px;
  width: 100%;
}

.item:hover,
.button:hover {
  cursor: pointer;
}

.item .category {
  padding-bottom: 6px;
}

.name {
  font-size: 16px;
  padding-left: 12px;
}

.category {
  padding-top: 6px;
  font-size: 12px;
}

/************* DETAILS ************/

.list {
  display: flex;
  flex-wrap: wrap;
}

.modal {
  position: fixed;
  top: 70px;
  left: 0px;
  -webkit-transform: translateX(-100%);
  transform: translateX(-100%);
  transition: all 0.3s ease-in-out;
  overflow: hidden;
  background: white;
  min-height: 100%;
}

.modal.opened {
  -webkit-transform: translateX(0);
  transform: translateX(0);
  transition: transform 0.3s ease-in-out;
  min-width: 100%;
  flex-direction: column;
}

.header .button {
  height: 70px;
  width: 70px;
  padding-left: 12px;
  padding-top: 17px;
  float: left;
}

.details-header {
  background-color: #34b379;
  height: 60px;
}

.details-header .name {
  position: relative;
  top: 13px;
}

.pad-16 {
  padding-top: 16px;
}

.pad-26 {
  padding-top: 26px;
}

/************* WEB **************/

.nav,
.footer {
  display: flex;
  justify-content: space-between;
}
.nav {
  width: 100px;
}

.web-body {
  position: relative;
  top: 70px;
  bottom: 50px;
  height: 1366px;
}

iframe {
  border: 0;
  height: 100%;
  width: 100%;
}

/************* FOOTER *************/

.footer {
  height: 50px;
  background-color: #2a2a2a;
  position: fixed;
  bottom: 0;
  right: 0;
  left: 0;
}

.footer,
.footer .button {
  font-size: 10px;
}

.footer .button {
  width: 50%;
  text-align: center;
}

.button img {
  padding-top: 5px;
  height: 26px;
}
.button div {
  padding: 2px 0px 2px 0px;
}

/********** RESPONSIVE *************/

@media only screen and (min-width: 520px) {
  .list-body {
    margin-bottom: 0px;
  }
  .footer {
    display: none;
  }
  .item {
    width: 50%;
  }
}

@media only screen and (min-width: 720px) {
  .item,
  .item img {
    height: 350px;
  }
  .item .name {
    padding: 0 0 6px 12px;
  }
}

@media only screen and (min-width: 1300px) {
  .item,
  .item img {
    height: 450px;
  }
}
