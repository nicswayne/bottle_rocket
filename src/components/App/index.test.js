import React from 'react';
import { shallow, render } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

import App from '.';

const mockStore = configureMockStore([thunk]);

it('renders without crashing.', () => {
    shallow(<App />);
});

it('matches snapshot', () => {
    const store = mockStore({});

    const wrapper = render(
        <Provider store={store}>
            <MemoryRouter>
                <App />
            </MemoryRouter>
        </Provider>
    );

    expect(wrapper).toMatchSnapshot();
});
