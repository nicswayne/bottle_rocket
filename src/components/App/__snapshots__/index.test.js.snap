// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class=""
  >
    <div>
      Header
    </div>
    <div>
      Body
      <div>
        Item
      </div>
    </div>
    <div
      class="footer"
    >
      <button
        class="button"
        type="button"
      >
        <img
          alt=""
          src="tab_lunch@2x.png"
        >
          
        </img>
        <div>
          lunch
        </div>
      </button>
      <button
        class="button"
        type="button"
      >
        <img
          alt=""
          src="tab_internets@2x.png"
        >
          
        </img>
        <div>
          internets
        </div>
      </button>
    </div>
  </div>
</div>
`;
